ocaml/xapi/xapi_clustering.ml

```diff
@@ -30,6 +30,22 @@ let with_clustering_lock f =
             f ())
          (fun () -> debug "Function execution finished; returned host-local clustering lock."))
 
+(** One of these cluster stacks should be configured and running *)
+let get_required_cluster_stacks ~__context ~sr_sm_type =
+  let sms_matching_sr_type =
+    Db.SM.get_records_where ~__context
+      ~expr:Db_filter_types.(Eq(Field "type", Literal sr_sm_type))
+  in
+  sms_matching_sr_type
+  |> List.map (fun (_sm_ref, sm_rec) -> sm_rec.API.sM_required_cluster_stack)
+  (* We assume that we only have one SM for each SR type, so this is only to satisfy type checking *)
+  |> List.flatten
+
+let with_clustering_lock_if_needed ~__context ~sr_sm_type f =
```

Comment:
Could you move this and the above function just above `find_cluster_host`? It doesn't seem to be really necessary to move `get_required_cluster_stacks` uphere.